assets/js/base/components/cart-checkout/product-details/index.tsx

```diff
@@ -30,20 +30,22 @@ const ProductDetails = ( {
 	return (
 		<ul className="wc-block-components-product-details">
 			{ details.map( ( detail ) => {
-				const className = detail.name
+				// Support both `key` and `name` props
+				detail.key = detail.key ? detail.key : detail.name || '';
```

Comment:
I would suggest using the nullish coalescing operator here, both for brevity and also to make sure intent is respected. I'ld also suggest not reassigning an object property when possible, to avoid side-effects. Perhaps not so important in this case, though.

So I'd probably go with something like:

```
const key = detail.key ?? detail.name ?? ''
```

Reply:
@sunyatasattva Using a nullish coalescing operator would use name if key was null, but what if it's empty? The behaviour in core we're copying uses the name if the key is empty (`null`, `false`, `0`, `''`).

Reply:
@mikejolley I see, thanks for the clarification. Indeed you are 100% spot on: the nullish coalescing would work for `null` and `undefined` only, which I thought was the intention here, sorry.

Given the interface below, however, it seems that `detail.key` and `detail.name` could only be `string` or `undefined`, so shouldn't TS save us from `false` and `0`?

Reply:
Yeah possibly, if respected. It won't save us from `''` or '`0`' or anything else which is considered empty.